### PR TITLE
Add new testing-utils library

### PR DESCRIPTION
**Pull request by reuk**
Splitting unit-test utilities into a separate library allows us to indicate which of the files in the `unit` directory are tests, and which are supporting files. It also allows us to re-use utilities from CBMC in dependent projects. In particular, the `load_java_class` function would be useful in some of the tests in test-gen.

Please would @thk123 and @LAJW review this?

### CODE REVIEW

File: unit/Makefile [hunk 1]
```diff
@@ -3,7 +3,6 @@
 # Source files for test utilities
 SRC = src/expr/require_expr.cpp \
       src/ansi-c/c_to_expr.cpp \
```
Comment (thk123) [inline unit/Makefile]: These files should probably also be moved into the `testing-utils`

File: unit/testing-utils/Makefile [hunk 2]
```diff
@@ -0,0 +1,16 @@
+SRC = \
+  load_java_class.cpp \
+  #
```
Comment (thk123) [inline unit/testing-utils/Makefile]: Nit: `# Empty last line` rather than just `#` as it looks like a typo

**Review by thk123 — CHANGES_REQUESTED** [review]
Looks good, but could you move the two other files in `unit/src` into this library (thanks for doing this btw!)

File: unit/Makefile (hunk 1 above)
Comment (reuk) [reply unit/Makefile]: Moved. It looks like none of the tests actually use these files though, is this correct?

**Comment by reuk** [comment]
Ready for re-review

**Review by LAJW — APPROVED** [review]
Look reasonable.

**Review by thk123 — APPROVED** [review]

**Pull request merged**